actions/src/email_actions.py

```diff
@@ -1,13 +1,34 @@
-from typing import Callable, Dict, List
+from dataclasses import dataclass
+from typing import Callable, Dict, List, Union
 from email_api.email_api import EmailApi
 from openstack_api.openstack_floating_ip import OpenstackFloatingIP
+from openstack_api.openstack_image import OpenstackImage
 from openstack_api.openstack_query import OpenstackQuery
 from openstack_api.openstack_server import OpenstackServer
 
 from st2common.runners.base_action import Action
 
 
 class EmailActions(Action):
+    @dataclass

```

Comment:
Can we extract this out to `lib/structs` (which should probably be called `lib/dataclasses` to align with Python instead of C)

That way it's clear this is shared between the lib and action layer, rather than being an implementation detail in just the action layer

Reply:
This isn't shared with the lib layer though, its only used in this file.

Reply:
Okay, then can we hoist it out of EmailActions into it's own class prefixed with `_` to `_EmailActionParams` to indicate it's a private implementation detail currently?
That way we don't have nested class structures within each module too.